tools/clang/lib/SPIRV/EmitVisitor.cpp

```diff
@@ -312,7 +339,8 @@ void EmitVisitor::initInstruction(SpirvInstruction *inst) {
     isGlobalVar = var->getStorageClass() != spv::StorageClass::Function;
   const auto op = inst->getopcode();
   emitDebugLine(op, inst->getSourceLocation(),
-                isGlobalVar ? &globalVarsBinary : &mainBinary);
+                isGlobalVar ? &globalVarsBinary : &mainBinary,
+                isa<SpirvDebugScope>(inst));
```

Comment:
suggestion: Maybe it's time that we change the `emitDebugLine` function signature to take a `SpirvInstruction`.
right now, we are passing:
* `op` (which is `inst->getopcode()`),
* `inst->getSourceLocation()`,
* global/main binary is decided based on `inst` storage class
* `isDebugScope` is decided based on `isa<SpirvDebugScope>(inst)`

so the new signature could be
```cpp
void emitDebugLine(SpirvInstruction *inst);
```
and it could do the logic inside the function.

Reply:
Had an offline chat. We concluded just keeping the current `emitDebugLine`. Thank you for your code review!